hoomd/VectorVariant.h

```diff
@@ -0,0 +1,289 @@
+
+    virtual ~VectorVariantBoxConstant() { }
+
+    /// Return the value.
+    virtual array_type operator()(uint64_t timestep)
+        {
+        return box_to_array(m_box);
+        }
+
+    std::shared_ptr<BoxDim> getBox()
+        {
+        return m_box;
+        }
+
+    void setBox(std::shared_ptr<BoxDim> box)
+        {
+        m_box = box;
+        }
+
+    protected:
+    std::shared_ptr<BoxDim> m_box;
+    };
+
+class PYBIND11_EXPORT VectorVariantBoxInterpolate : public VectorVariantBox
+    {
+    public:
+    /** Construct a VectorVariantBoxInterpolate to interpolate between two boxes linearly in time.
+
+        @param box1 The initial box
+        @param box2 The final box
+    */
+    VectorVariantBoxInterpolate(std::shared_ptr<BoxDim> box1,
+                                std::shared_ptr<BoxDim> box2,
+                                std::shared_ptr<Variant> variant)
+        : m_box1(box1), m_box2(box2), m_variant(variant)
+        {
+        }
+
+    /// Return the value.
+    virtual array_type operator()(uint64_t timestep)
+        {
+        Scalar min = m_variant->min();
+        Scalar max = m_variant->max();
+        Scalar cur_value = (*m_variant)(timestep);
+        Scalar scale = 0;
+        if (cur_value == max)
+            {
+            scale = 1;
+            }
+        else if (cur_value > min)
+            {
+            scale = (cur_value - min) / (max - min);
+            }
+
+        const auto& box1 = *m_box1;
+        const auto& box2 = *m_box2;
+        Scalar3 new_L = box2.getL() * scale + box1.getL() * (1.0 - scale);
+        Scalar xy = box2.getTiltFactorXY() * scale + (1.0 - scale) * box1.getTiltFactorXY();
+        Scalar xz = box2.getTiltFactorXZ() * scale + (1.0 - scale) * box1.getTiltFactorXZ();
+        Scalar yz = box2.getTiltFactorYZ() * scale + (1.0 - scale) * box1.getTiltFactorYZ();
+        array_type value = {new_L.x, new_L.y, new_L.z, xy, xz, yz};
+        return value;
+        }
+
+    std::shared_ptr<BoxDim> getBox1()
+        {
+        return m_box1;
+        }
+
+    void setBox1(std::shared_ptr<BoxDim> box)
+        {
+        m_box1 = box;
+        }
+
+    std::shared_ptr<BoxDim> getBox2()
+        {
+        return m_box2;
+        }
+
+    void setBox2(std::shared_ptr<BoxDim> box)
+        {
+        m_box2 = box;
+        }
+
+    /// Set the variant for interpolation
+    void setVariant(std::shared_ptr<Variant> variant)
+        {
+        m_variant = variant;
+        }
+
+    /// Get the variant for interpolation
+    std::shared_ptr<Variant> getVariant()
+        {
+        return m_variant;
+        }
+
+    protected:
+    /// The starting box, associated with the minimum of the variant.
+    std::shared_ptr<BoxDim> m_box1;
+
+    /// The final box, associated with the maximum of the variant.
+    std::shared_ptr<BoxDim> m_box2;
+
+    /// Variant that interpolates between boxes.
+    std::shared_ptr<Variant> m_variant;
+    };
+
```

Comment:
Class level overview documentation for this and the other classes would be useful.